### PR TITLE
fix: Check for old sparse vector qdrant.base.py

### PR DESCRIPTION
**Pull request by Anush008**
# Description
This PR fixes the check used for sparse vectors built using the old encoder.

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py [hunk 1]
```diff
@@ -874,7 +874,7 @@ def use_old_sparse_encoder(self, collection_name: str) -> bool:
         return (
             self._collection_exists(collection_name)
             and SPARSE_VECTOR_NAME_OLD
-            in self.client.get_collection(collection_name).config.params.vectors
+            in self.client.get_collection(collection_name).config.params.sparse_vectors
```
[the remark this answers is not in the document]
Comment (semoal) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py:877]: I would do:

```python
    def use_old_sparse_encoder(self, collection_name: str) -> bool:
        collection = self.client.get_collection(collection_name)
        return (
            self._collection_exists(collection_name)
            and collection.config.params.sparse_vectors is not None
            and SPARSE_VECTOR_NAME_OLD in collection.config.params.sparse_vectors
        )
```

[the remark this answers is not in the document]
Comment (Anush008) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py]: https://github.com/run-llama/llama_index/pull/12916/commits/1a5fefa05e8830898266cc0341eb080cc0ae8d0d

[the remark this answers is not in the document]
Comment (Anush008) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py]: I am still missing why the code execution would get here if the collection didn't have any sparse vectors. Was `enable_hybrid` set on such a collection?

[the remark this answers is not in the document]
Comment (semoal) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-qdrant/llama_index/vector_stores/qdrant/base.py]: Yes

**Review by logan-markewich — APPROVED** [review]

**Comment by jack-gits** [comment]
is this issue fixed in the 0.10.35?

I still met this issue in the newest version. below is the detail information
`
llama_cpp_python                         0.2.56
llama-hub                                0.0.77
llama-index                              0.10.35
llama-index-agent-openai                 0.1.5
llama-index-cli                          0.1.9
llama-index-core                         0.10.35.post1
llama-index-embeddings-azure-openai      0.1.9
llama-index-embeddings-openai            0.1.6
llama-index-indices-managed-llama-cloud  0.1.4
llama-index-legacy                       0.9.48
llama-index-llms-azure-openai            0.1.7
llama-index-llms-openai                  0.1.16
llama-index-multi-modal-llms-openai      0.1.4
llama-index-program-openai               0.1.4
llama-index-question-gen-openai          0.1.3
llama-index-readers-file                 0.1.11
llama-index-readers-llama-parse          0.1.3
llama-index-vector-stores-qdrant         0.2.5
llama-parse                              0.3.9
llamaindex-py-client                     0.1.19
qdrant-client                            1.9.1
`

Exception information:

`2024-05-09 10:26:05.551 Uncaught app exception
Traceback (most recent call last):
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 600, in _run_script
    exec(code, module.__dict__)
  File "/Users/xuhao/work/SourceCode/neuron/document-app-demo/document-app/pages/Search Assistant.py", line 39, in <module>
    results = qdrantService.do_fusion_search_documents(query, 10,prev_next_node=True,with_summary=True)
  File "/Users/xuhao/work/SourceCode/neuron/document-app-demo/document-app/customization/qdrant_service.py", line 427, in do_fusion_search_documents
    pydanticResponse= query_engine.query(query)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/instrumentation/dispatcher.py", line 274, in wrapper
    result = func(*args, **kwargs)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/base/base_query_engine.py", line 53, in query
    query_result = self._query(str_or_query_bundle)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/instrumentation/dispatcher.py", line 274, in wrapper
    result = func(*args, **kwargs)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/query_engine/retriever_query_engine.py", line 189, in _query
    nodes = self.retrieve(query_bundle)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/query_engine/retriever_query_engine.py", line 144, in retrieve
    nodes = self._retriever.retrieve(query_bundle)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/instrumentation/dispatcher.py", line 274, in wrapper
    result = func(*args, **kwargs)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/base/base_retriever.py", line 244, in retrieve
    nodes = self._retrieve(query_bundle)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/retrievers/fusion_retriever.py", line 263, in _retrieve
    results = self._run_sync_queries(queries)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/retrievers/fusion_retriever.py", line 251, in _run_sync_queries
    results[(query.query_str, i)] = retriever.retrieve(query)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/instrumentation/dispatcher.py", line 274, in wrapper
    result = func(*args, **kwargs)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/base/base_retriever.py", line 244, in retrieve
    nodes = self._retrieve(query_bundle)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/instrumentation/dispatcher.py", line 274, in wrapper
    result = func(*args, **kwargs)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/indices/vector_store/retrievers/retriever.py", line 101, in _retrieve
    return self._get_nodes_with_embeddings(query_bundle)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/core/indices/vector_store/retrievers/retriever.py", line 177, in _get_nodes_with_embeddings
    query_result = self._vector_store.query(query, **self._kwargs)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/llama_index/vector_stores/qdrant/base.py", line 505, in query
    sparse_response = self._client.search_batch(
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/qdrant_client/qdrant_client.py", line 251, in search_batch
    return self._client.search_batch(
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/qdrant_client/qdrant_remote.py", line 408, in search_batch
    http_res: List[List[models.ScoredPoint]] = self.http.points_api.search_batch_points(
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/qdrant_client/http/api/points_api.py", line 1354, in search_batch_points
    return self._build_for_search_batch_points(
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/qdrant_client/http/api/points_api.py", line 568, in _build_for_search_batch_points
    return self.api_client.request(
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/qdrant_client/http/api_client.py", line 76, in request
    return self.send(request, type_)
  File "/Users/xuhao/anaconda3/envs/python3.9/lib/python3.9/site-packages/qdrant_client/http/api_client.py", line 99, in send
    raise UnexpectedResponse.for_response(response)
qdrant_client.http.exceptions.UnexpectedResponse: Unexpected Response: 400 (Bad Request)
Raw response content:
b'{"status":{"error":"Wrong input: Vector params for text-sparse-new are not specified in config"},"time":0.000037773}'
`

**Comment by Anush008** [comment]
@jack-gits, could you try re-creating the collection?

**Comment by logan-markewich** [comment]
@jack-gits the vector store version is separate. You'd want to do `pip install -U llama-index-vector-stores-qdrant`

**Comment by jack-gits** [comment]
already installed " llama-index-vector-stores-qdrant         0.2.5", tried the the 0.2.8 version as well, it raised an error same.

**Comment by jack-gits** [comment]
@Anush008  current collection already has the text-sparse. still need to re-create the collection with new version llama-index?
image

**Pull request merged**